File: yggdrasil/examples/model_function/src/model_function_modelB.c
#include <stdio.h>

int model_function(char *inputB, uint64_t length_inputB,
		   char** outputB, uint64_t* length_outputB) {

  length_outputB[0] = length_inputB;
  outputB[0] = (char*)malloc(length_inputB);
  memcpy(outputB[0], inputB, length_inputB);
  outputB[0][length_inputB] = '\0';
  printf("Model B: %s (length = %d)\n", *outputB, (int)(*length_outputB));
  return 0;

}

File: yggdrasil/examples/model_function/src/model_function_modelB.f90
module example_module
contains
  function model_function(inputB, outputB) result(out)
    character(len=*), intent(in) :: inputB
    character(len=:), pointer :: outputB
    logical :: out
    out = .true.
    allocate(character(len=len(inputB)) :: outputB)
    outputB = inputB
    write(*, '("Model B: ",A," (length = ",I3,")")') outputB, len(outputB)
  end function model_function
end module example_module
